src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfoExtensions.cs

```diff
@@ -45,7 +51,18 @@ public static ProjectInfo CreateProjectInfo(this ProjectFileInfo projectFileInfo
                 language: LanguageNames.CSharp,
                 filePath: projectFileInfo.FilePath,
                 outputFilePath: projectFileInfo.TargetPath,
-                compilationOptions: projectFileInfo.CreateCompilationOptions());
+                compilationOptions: projectFileInfo.CreateCompilationOptions(),
+                analyzerReferences: analyzerReferences);
+        }
+
+        private static IEnumerable<AnalyzerReference> ResolveAnalyzerReferencesForProject(ProjectFileInfo projectFileInfo, IAnalyzerAssemblyLoader analyzerAssemblyLoader)
+        {
+            foreach(var analyzerAssemblyPath in projectFileInfo.Analyzers.Distinct())
```

Comment:
we also have the possibility of loading code refactorings via global configuration https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Abstractions/Options/RoslynExtensionsOptions.cs#L10. 

At the moment only refactorings work this way, but not analyzers - it would make sense to include assemblies discovered this way in analyzers

Reply:
If i remember correctly (implemented like 6 months back 🙂 ) it should be implemented in current version:

```
                var allAnalyzers = _providers
                    .SelectMany(x => x.CodeDiagnosticAnalyzerProviders)
                    .Concat(project.AnalyzerReferences.SelectMany(x => x.GetAnalyzers(project.Language)))
                    .ToImmutableArray();
```

_providers is implementation of ICodeActionProvider which is extended to support analyzers and code fixes.

Reply:
ah you are right, brilliant 👍the PR is difficult to view on github at this point 😄